### PR TITLE
fix: catch and return false if getTypeFromTypeNode fails

### PR DESCRIPTION
**Pull request by wondersloth**
Fixes: #1060 

Validated locally.

- [x] Needs a test.

### CODE REVIEW

File: packages/ts-utils/src/helpers/typescript-ast.ts [hunk 1]
```diff
@@ -129,7 +129,14 @@ export function isVariableOfCatchClause(node: Node): boolean {
 
 export function canTypeBeResolved(checker: TypeChecker, typeNode: TypeNode): boolean {
   if (isTypeReferenceNode(typeNode)) {
```
Comment (egorio) [inline packages/ts-utils/src/helpers/typescript-ast.ts:131]: You could change this if statement to 

```
if (isTypeReferenceNode(typeNode)) {
    const typeArguments = typeNode.typeArguments || [];

    const isTypeNodeStrict = (checker: TypeChecker, typeNode: TypeNode): boolean => {
      try {
        // Try to resolve TypeNode to Type... it possible to trigger and exception in some weired cases
        const type = checker.getTypeFromTypeNode(typeNode);

        // Check if TypeNode is resolved to a meaningful Type, not any
        // Also can be validated with (type as unknown as { intrinsicName?: string }).intrinsicName === 'error';
        return type.flags !== ts.TypeFlags.Any;
      } catch (e) {
        return false
      }
    };
    
    return isTypeNodeStrict(checker, typeNode) && !typeArguments.find((node) => !canTypeBeResolved(checker, node));
  }
  ```
  
  Will be easy to read

**Comment by egorio** [comment]
Test case

```
/**
 * The return type is throwing a error
 * @param {[number, number]} ab
 * @param {Array<string>} c
 * @return {Array<{content: string}>}
 */
export function moreCases([a, b], c) {
  a; b; c;
  return [];
}
```

add to packages/codefixes/test/fixtures/base-codefixes/annotateWithStrictTypeFromJSDoc/index.ts

**Review by egorio — CHANGES_REQUESTED** [review]
Test case

```
/**
 * The return type is throwing a error
 * @param {[number, number]} ab
 * @param {Array<string>} c
 * @return {Array<{content: string}>}
 */
export function moreCases([a, b], c) {
  a; b; c;
  return [];
}
```

add to packages/codefixes/test/fixtures/base-codefixes/annotateWithStrictTypeFromJSDoc/index.ts

**Review by wondersloth — COMMENTED** [review]
Looks good. Thank @egorio for improving the solution and adding a test.

**Review by egorio — APPROVED** [review]

File: packages/codefixes/test/__snapshots__/base-codefixes.test.ts.snap [hunk 2]
```diff
@@ -563,6 +563,39 @@ export function think(
...
   return a;
 }
+
+/**
+ * The return type is throwing an error
+ * @param {[number, number]} a-and-b
+ * @param {Array<string>} c
+ * @return {Array<{content: string}>}
+ */
+// @ts-expect-error @rehearsal TODO TS7031: Binding element 'a' implicitly has an 'any' type.
```
Comment (egorio) [inline packages/codefixes/test/__snapshots__/base-codefixes.test.ts.snap:573]: This works in the next function, but not here.

Created a ticket for this [case](https://github.com/rehearsal-js/rehearsal-js/issues/1062)

**Pull request merged**